### PR TITLE
[RDY] Vim 7.4.718

### PR DESCRIPTION
**Pull request by brcolow**
vim-patch:7.4.718

Problem:    Autocommands triggered by quickfix cannot get the current title
            value.
Solution:   Set w:quickfix_title earlier. (Yannick)
            Also move the check for a title into the function.

https://github.com/vim/vim/commit/81278efadf45ad745ec9733fdd9970f070571544

### CODE REVIEW

**Comment by Shougo** [comment]
LGTM

File: src/nvim/quickfix.c [hunk 1]
```diff
@@ -2274,10 +2268,12 @@ static void qf_update_buffer(qf_info_T *qi)
...
 }
 
+// Set "w:quickfix_title" if "qi" has a title.
 static void qf_set_title_var(qf_info_T *qi)
 {
-  set_internal_string_var((char_u *)"w:quickfix_title",
-      qi->qf_lists[qi->qf_curlist].qf_title);
+  if (qi->qf_lists[qi->qf_curlist].qf_title != NULL) {
+      set_internal_string_var((char_u *)"w:quickfix_title",
+                                  qi->qf_lists[qi->qf_curlist].qf_title);
```
Comment (Pyrohh) [inline src/nvim/quickfix.c]: missing closing brace

Comment (brcolow) [inline src/nvim/quickfix.c]: Nice catch, fixed, thanks a lot.

**Comment by Pyrohh** [comment]
It seems that the linter isn't happy.

**Comment by brcolow** [comment]
Should be good to go now @Pyrohh

**Comment by brcolow** [comment]
Updated to reflect the suggestion by @oni-link and the confirmation by @chrisbra.

File: src/nvim/quickfix.c [hunk 2]
```diff
@@ -2274,10 +2267,13 @@ static void qf_update_buffer(qf_info_T *qi)
...
 }
 
+// Set "w:quickfix_title" if "qi" has a title.
 static void qf_set_title_var(qf_info_T *qi)
 {
-  set_internal_string_var((char_u *)"w:quickfix_title",
-      qi->qf_lists[qi->qf_curlist].qf_title);
+  if (qi->qf_lists[qi->qf_curlist].qf_title != NULL) {
+      set_internal_string_var((char_u *)"w:quickfix_title",
+                              qi->qf_lists[qi->qf_curlist].qf_title);
```
Comment (oni-link) [inline src/nvim/quickfix.c]: I would try to make it a bit more readable:
```c
char_u * const title = qi->qf_lists[qi->qf_curlist].qf_title;
if (title) {
  set_internal_string_var((char_u *)"w:quickfix_title", title);
}
```

**Comment by oni-link** [comment]
LGTM.

**Pull request merged**